app/store/thunk.js

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {
+    captureMessage,
+    cleanUrlForLogging,
+    LOGGER_JAVASCRIPT_WARNING,
+} from 'app/utils/sentry';
+
+// Creates middleware that mimics thunk while catching network errors thrown by Client4 that haven't
+// been otherwise handled.
+export function createThunkMiddleware() {
```

Comment:
Interesting thing to note is that outside of the special promise handling here, this is basically the entire code for thunk. It's a brilliantly simple library.